### PR TITLE
Automates server deployment

### PR DESCRIPTION
**Pull request by crdant**
TL;DR
-----

Uses Terraform to automatically deploy the server

Details
-------

Implements Terraform automation to create the new server from
Ubuntu's cloud image for Focal. The Terraform templates deploy
from the daily build OVA and generate the `cloud-init` user 
data using the Carvel Terraform provider.

The automation uses the same user data templates that the 
manual approach uses to generate user data, but reclocated
the template directory under the `src` directory where the 
Terraform templates are also located.

Terreform state is stored in Google Cloud Storage. There's a
script `prepare-remote-state` included to get the bucket and
service account ready. A parallel script, `prepare-vcenter`
configures a vCenter user for the Terraform automation. Right
now that users is an administrator to simplify set up.

### CODE REVIEW

**Comment by crdant** [comment]
Update README before merging.
